src/App.jsx

```diff
@@ -14,33 +16,97 @@ export const goodsFromServer = [
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SortType = {
+  NONE: 'NONE',
+  ALPHABET: 'ALPHABET',
+  LENGTH: 'LENGTH',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [isReversed, setReversed] = React.useState(false);
+  const [sortType, setSortType] = React.useState(SortType.NONE);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  // eslint-disable-next-line no-shadow
+  function getReorderedGoods(goods, { sortType, isReversed }) {
+    const visibleGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    switch (sortType) {
+      case SortType.ALPHABET:
+        visibleGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.LENGTH:
+        visibleGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  }
+
+  const resetFunction = () => {
+    setSortType(SortType.NONE);
+    setReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={() => setSortType(SortType.ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+          onClick={() => setSortType(SortType.LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReversed(reverse => !reverse)}
+        >
+          Reverse
+        </button>
+
+        {(isReversed || sortType !== SortType.NONE) && (
+          <button
+            type="button"
+            className={classNames('button', 'is-danger', 'is-light')}
+            onClick={resetFunction}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {getReorderedGoods(goodsFromServer, { sortType, isReversed }).map(
+          (good, index) => (
+            // eslint-disable-next-line react/no-array-index-key
```

Comment:
Using the index as a key in list rendering can lead to performance issues and unexpected behavior, especially if the list changes dynamically. Consider using a unique identifier for each item if possible.